forge-gui-mobile/src/forge/adventure/data/AdventureQuestStage.java

```diff
@@ -325,9 +325,14 @@ public AdventureQuestController.QuestStatus handleEvent(AdventureQuestEvent even
                 }
                 break;
             case Fetch:
-                status = event.type == AdventureQuestEventType.RECEIVEITEM
-                        && (itemNames.isEmpty()) || (event.item != null && itemNames.contains(event.item.name))
-                        && ++progress1 >= count1 ? COMPLETE : status;
+                if (event.type == AdventureQuestEventType.RECEIVEITEM) {
+                    if ((itemNames.isEmpty()) || (event.item != null && itemNames.contains(event.item.name)))
+                        status = ++progress1 >= count1 ? COMPLETE : status;
+                }
+                else if (event.type == AdventureQuestEventType.USEITEM) {
+                    if ((itemNames.isEmpty()) || (event.item != null && itemNames.contains(event.item.name)))
+                        status = ++progress3 >= count3 ? COMPLETE : status;
```

Comment:
I'm a little bit worried about how this whole class interacts with its different progress/count variables. But let's try it and keep an eye on reports.